src/main/java/com/amplitude/api/IdentifyInterceptor.java

```diff
@@ -0,0 +1,193 @@
+        try {
+            List<JSONObject> identifys = dbHelper.getIdentifyInterceptors(lastIdentifyInterceptorId, -1);
+            if (identifys.isEmpty()) {
+                return null;
+            }
+            JSONObject identifyEvent = identifys.get(0);
+            JSONObject identifyEventUserProperties = identifyEvent.getJSONObject("user_properties");
+            JSONObject userProperties = identifyEventUserProperties .getJSONObject(Constants.AMP_OP_SET);
+            for (int i = 1; i < identifys.size(); i++) {
+                JSONObject identify = identifys.get(i);
+                JSONObject setUserProperties = identify.getJSONObject("user_properties")
+                        .getJSONObject(Constants.AMP_OP_SET);
+                Iterator<?> keys = setUserProperties.keys();
+                while (keys.hasNext()) {
+                    String key = (String) keys.next();
+                    userProperties.put(key, setUserProperties.get(key));
+                }
+            }
+            identifyEventUserProperties.put(Constants.AMP_OP_SET, userProperties);
+            identifyEvent.put("user_properties", identifyEventUserProperties);
+            dbHelper.removeIdentifyInterceptors(lastIdentifyInterceptorId);
+            return identifyEvent;
+        } catch (JSONException e) {
+            AmplitudeLog.getLogger().w(TAG, "Identify Merge error: " + e.getMessage());
+        }
+        return null;
+    }
+
+    private void scheduleTransfer() {
+        if (transferScheduled.getAndSet(true)) {
+            return;
+        }
+
+        logThread.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                transferScheduled.set(false);
+                transferInterceptedIdentify();
+            }
+        }, identifyBatchIntervalMillis);
+    }
+
+    public void transferInterceptedIdentify() {
+        JSONObject identifyEvent = getTransferIdentifyEvent();
+        if (identifyEvent == null) {
+            return;
+        }
+        client.saveEvent(Constants.IDENTIFY_EVENT, identifyEvent);
+    }
+
+    private JSONObject fetchAndMergeToNormalEvent(JSONObject event) {
+        try {
+            List<JSONObject> identifys = dbHelper.getIdentifyInterceptors(lastIdentifyInterceptorId, -1);
+            if (identifys.isEmpty()) {
+                return event;
+            }
+            JSONObject userProperties = event.getJSONObject("user_properties");
+            for (JSONObject identify : identifys) {
+                JSONObject setUserProperties = identify.getJSONObject("user_properties")
+                        .getJSONObject(Constants.AMP_OP_SET);
+                Iterator<?> keys = setUserProperties.keys();
+                while (keys.hasNext()) {
+                    String key = (String) keys.next();
+                    userProperties.put(key, setUserProperties.get(key));
+                }
+            }
+            event.put("user_properties", userProperties);
```

Comment:
Same here, I think we should create a new user_properties dictionary and apply the intercepted identify values, then the event user_properties.

Reply:
Oh, I see your point, the order matters. That makes sense, I will update it.